halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/kubernetes/v1/KubernetesV1DistributedService.java

```diff
@@ -417,24 +417,8 @@ default void resizeVersion(
     return getObjectMapper().convertValue(description, new TypeReference<Map<String, Object>>() {});
   }
 
-  default KubernetesContainerDescription buildContainer(
-      String name,
-      ServiceSettings settings,
-      List<ConfigSource> configSources,
-      DeploymentEnvironment deploymentEnvironment,
-      DeployKubernetesAtomicOperationDescription description) {
-    KubernetesContainerDescription container = new KubernetesContainerDescription();
-    KubernetesProbe readinessProbe = new KubernetesProbe();
+  default KubernetesHandler buildProbeHandler(int port, String scheme, String healthEndpoint) {
     KubernetesHandler handler = new KubernetesHandler();
-    int port = settings.getPort();
-    String scheme = settings.getScheme();
-    if (StringUtils.isNotEmpty(scheme)) {
-      scheme = scheme.toUpperCase();
-    } else {
-      scheme = null;
-    }
-
-    String healthEndpoint = settings.getHealthEndpoint();
     if (healthEndpoint != null) {
```

Comment:
This looks like a bigger diff than it is, really I just plopped all the probe handler stuff into a separate method